packages/lib-classifier/src/components/Classifier/components/SubjectViewer/components/InteractionLayer/InteractionLayerContainer.js

```diff
@@ -28,36 +24,51 @@ class InteractionLayerContainer extends Component {
     this.onPointerDown = this.onPointerDown.bind(this)
     this.onPointerMove = this.onPointerMove.bind(this)
     this.onPointerUp = this.onPointerUp.bind(this)
+    this.state = {
+      activeMark: null
+    }
   }
 
-  componentDidMount () {
-    this.setCurrentSVG()
-  }
+  convertEvent (event) {
+    const type = event.type
 
-  componentDidUpdate (prevProps) {
-    if (this.props.svg !== prevProps.svg) {
-      this.setCurrentSVG()
+    const clientX = event.clientX
+    const clientY = event.clientY
+    const svgEventOffset = this.getEventOffset(clientX, clientY)
+
+    const svgCoordinateEvent = {
+      pointerId: event.pointerId,
+      type,
+      x: svgEventOffset.x,
+      y: svgEventOffset.y
     }
+
+    return svgCoordinateEvent
   }
 
-  setCurrentSVG () {
-    const { setCreateSVGPoint, setGetScreenCTMInverse, svg } = this.props
-    if (svg) {
-      setCreateSVGPoint(svg.createSVGPoint())
-      setGetScreenCTMInverse(svg.getScreenCTM().inverse())
-    }
+  getEventOffset (x, y) {
+    const { svg } = this.props
+    const svgEvent = svg.createSVGPoint()
+    svgEvent.x = x
+    svgEvent.y = y
+    const svgEventOffset = svgEvent.matrixTransform(svg.getScreenCTM().inverse())
+
+    return svgEventOffset
   }
 
   onPointerDown (event) {
-    this.props.addToStream(event)
+    const { createMark } = this.props
+    const activeMark = createMark(this.convertEvent(event))
+    this.setState({ activeMark })
   }
-
+  
   onPointerMove (event) {
-    this.props.addToStream(event)
+    const { activeMark } = this.state
+    activeMark && activeMark.setCoordinates(this.convertEvent(event))
   }
 
   onPointerUp (event) {
-    this.props.addToStream(event)
+    this.setState({ activeMark: null })
```

Comment:
Related to the comment I've made about the display of the delete button. What does 'active' mean? That it is selected and can be edited? This is true when you first create the mark with the PFE interaction.

Reply:
Good point. If we follow PFE, there should always be one mark that's focussed and editable.

Reply:
So, I deleted this line and now the initial drawing gesture never completes. A point continues to follow the pointer even after pointerup. 🤔

Reply:
`activeMark` here is a mark that is in the process of being created but the drawing tools also use `active` to switch styling ie. an active mark is one that can be edited or deleted.